File: model/src/main/java/me/rolgalan/marvelw/model/PaginableList.java
package me.rolgalan.marvelw.model;

import java.util.List;

/**
 * Created by Roldán Galán on 20/11/2017.
 */

public interface PaginableList<K> extends List<K> {

    int getTotalPossibleSize();

    void setTotalPossibleSize(int totalPossibleSize);

    int getOffset();

    void setOffset(int offset);

    boolean hasMoreItemsToLoad();

}

File: model/src/main/java/me/rolgalan/marvelw/model/PaginableList.kt
package me.rolgalan.marvelw.model

/**
 * Created by Roldán Galán on 20/11/2017.
 */

interface PaginableList<K> : MutableList<K> {

    var totalPossibleSize: Int

    var offset: Int

    fun hasMoreItemsToLoad(): Boolean

}
